### DISCUSSION TITLE
Spy blocker?

### DISCUSSION THREAD

**Discussion by fabcard** [farag2/Sophia-Script-for-Windows#257] [Q&A] [answered]
Hello everyone,
Sorry if this was already answered before. If yes, feel free to delete this discussion.
So does Sophia have any kind of firewall/hosts IP spy block?
I use for a long time the spy version of  WindowsSpyBlocker from @crazy-max https://github.com/crazy-max/WindowsSpyBlocker. I really like it, and they are always updating it too.
I'm wondering if it's possible to make Sophia install those IP's and make it run a task to check for updates weekly. Maybe showing on display while Sophia is running which version of the spy blocker we want to install: the spy, extra and update, just like it does for the apps removal, services, etc.
Thanks in advance.

**Comment by farag2** [comment, marked as answer]
> So does Sophia have any kind of firewall/hosts IP spy block?

No. And will never have.

> I use for a long time the spy version of WindowsSpyBlocker

What's the purpose?..

> I'm wondering if it's possible to make Sophia install those IP's


I will never intergrate such feature due to its' dengerousness. But if it's simple, I may try to add a new function for you. Please describe how does it work ( I mean, we donwload the app, install and and then by a task it checks the updates, ot how?).

**Comment by fabcard** [comment]
Hello @farag2 ,
1- Well, sorry. I though this could be a good idea.
2- "WindowsSpyBlocker 🛡️ is an application written in Go and delivered as a single executable to block spying and tracking on Windows systems."
3-I'm not an expert, but I believe there's nothing dangerous on that. It's just a list of IP's that Windows try to use to send info to Microsoft.
You can install it via their "WindowsSpyBlocker.exe" from releases, or can set them manually like on hosts or windows firewall
![1](https://user-images.githubusercontent.com/88634350/135828995-46b44fa8-0127-4261-b20b-cb4cdda5410e.jpg)
![2](https://user-images.githubusercontent.com/88634350/135828985-39e3455f-9530-4a75-b545-5cfe3053f9b9.jpg)
They have list compatible with other programs too like Kaspersky, Eset, simplewall, etc.
My idea was to make Sophia to add those IP's to block, or make a new script to install IP's on windows firewall, like pics above or on hosts file. Then add a task scheduler to run every week to check if there's update for the IP lists.
There are 3 versions of lists. One is basic, the other blocks a few other extras and last one also block windows updates.
If you are interested, you can check https://github.com/crazy-max/WindowsSpyBlocker
Here is an example for hosts https://raw.githubusercontent.com/crazy-max/WindowsSpyBlocker/master/data/hosts/spy.txt
And here for windows firewall https://raw.githubusercontent.com/crazy-max/WindowsSpyBlocker/master/data/firewall/spy.txt
Thank you!
